libid/ui/cmdfiles.cpp

```diff
@@ -3759,8 +3760,35 @@ static CmdArgFlags cmd_xy_shift(const Command &cmd)
     return CmdArgFlags::FRACTAL_PARAM | CmdArgFlags::PARAM_3D;
 }
 
+// tolerance=?
+static CmdArgFlags cmd_tolerance(const Command &cmd)
+{
+    g_perturbation_tolerance = cmd.float_vals[0];
+    return CmdArgFlags::NONE;
+}
+
+// perturbation=?
+static CmdArgFlags cmd_perturbation(const Command &cmd)
+{
+    std::string p = cmd.value;
+    if (p == "auto")
+    {
+        g_perturbation = PerturbationMode::AUTO;
```

Comment:
Instead of adding another global variable, add setter/getter functions and encapsulate the state as a static variable or member of the PertEngine class.